rest_framework/views.py

```diff
@@ -90,10 +90,10 @@ def exception_handler(exc, context):
         if getattr(exc, 'wait', None):
             headers['Retry-After'] = '%d' % exc.wait
 
-        if isinstance(exc.detail, (list, dict)):
+        if isinstance(exc.detail, (list, tuple, dict)):
             data = exc.detail
         else:
-            data = {'detail': exc.detail}
+            data = {'detail': exc.detail, 'code': exc.detail.code}
```

Comment:
Per your suggestion, I might add `code` support as a separate exception handler. That said, I'd introduce it in a new PR.